textile/objects-features.textile

```diff
@@ -0,0 +1,145 @@
+---
+title: Objects Features
+section: client-lib-development-guide
+index: 65
+jump_to:
+  Help with:
+    - Objects Features Overview#overview
+---
+
+h2(#overview). Overview
+
+This document outlines the feature specification for the Objects feature of the Realtime system. It is currently under development and stored separately from the main specification to simplify the initial implementation of the feature in other SDKs. Once completed, it will be moved to the main "features":../features spec.
+
+Objects feature enables clients to store shared data as "objects" on a channel. When an object is updated, changes are automatically propagated to all subscribed clients in realtime, ensuring each client always sees the latest state.
+
+h3(#realtime-objects). RealtimeObjects
+
+* @(RTO1)@ @Objects#getRoot@ function:
+** @(RTO1a)@ Requires the @OBJECT_SUBSCRIBE@ channel mode to be granted per "RTO2":#RTO2
+** @(RTO1b)@ If the channel is in the @DETACHED@ or @FAILED@ state, the library should indicate an error with code 90001
```

Comment:
What if the channel is initialised, detaching, suspended, etc?

I think it would be better to say "If the channel is not ATTACHED[ or ATTACHING]...", since these are probably the only two cases where we would expect the operation to work. If we want it to work when ATTACHING we need to spec how the operation will be queued.

The other thing we need to spec is whether the operation should implicitly attach an INITIALIZED channel